crates/api/src/site/mod_log.rs

```diff
@@ -44,19 +40,17 @@ pub async fn get_mod_log(
   let type_ = data.type_.unwrap_or(All);
   let community_id = data.community_id;
 
-  let (local_person_id, is_admin) = match local_user_view {
-    Some(s) => (s.person.id, is_admin(&s).is_ok()),
-    None => (PersonId(-1), false),
-  };
-  let community_id_value = match community_id {
-    Some(s) => s,
-    None => CommunityId(-1),
-  };
-  let is_mod_of_community = data.community_id.is_some()
-    && is_mod_or_admin(&mut context.pool(), local_person_id, community_id_value)
-      .await
-      .is_ok();
-  let hide_modlog_names = local_site.hide_modlog_mod_names && !is_mod_of_community && !is_admin;
+  let mut is_mod_of_community = false;
+  let mut is_admin_ = false;
+  if let Some(local_user_view) = local_user_view {
+    is_mod_of_community =
+      check_community_mod_action_opt(&local_user_view, community_id, &mut context.pool())
+        .await
+        .is_ok()
+        && community_id.is_some();
+    is_admin_ = is_admin(&local_user_view).is_ok();
+  }
+  let hide_modlog_names = local_site.hide_modlog_mod_names && !is_mod_of_community && !is_admin_;
```

Comment:
This is a bit confusing, please check that its still doing the right thing.

Reply:
looks ok to me. I don't think there's a reason to have two separate flags so it'd be cleaner (without mutables) as `let is_mod_or_admin = if let Some(...) = ... { check_mod(...) || is_admin(...) } else { false }`

Reply:
Makes sense, changed